fdbserver/DataDistribution.actor.cpp

```diff
@@ -3713,9 +3716,106 @@ static std::set<int> const& normalDataDistributorErrors() {
 	return s;
 }
 
+ACTOR Future<Void> ddSnapCreate(DistributorSnapRequest snapReq, Reference<AsyncVar<struct ServerDBInfo>> db ) {
+	state Database cx = openDBOnServer(db, TaskDefaultEndpoint, true, true);
+	state double snapTimeout = g_network->isSimulated() ? 10.0 : SERVER_KNOBS->SNAP_CREATE_MAX_TIMEOUT;
```

Comment:
> Instead I would suggest to wrap the whole ddSnapCreate actor into a timeoutError call. It would also simplify the code.
yes, makes sense. 
I think even that is not needed, because the caller is already calling under a timeoutError and i think if that timesout then it is going to cancel all these actors.

Reply:
> I think it would be cleaner to move this check for simulated network into the initialization code in `fdbserver/Knobs.cpp`.

I tried it and this would not work because g_network is not available at initialization, that is possible the reason why in many places - similar style is followed.

Reply:
ah yes, this makes sense. never mind then